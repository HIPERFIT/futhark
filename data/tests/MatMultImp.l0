fun *[[int]] matmultImp([[int]] a, [[int]] b) =
    let N   = size(0, a)            in
    let res = replicate(N, iota(N)) in
    loop (res) = for i < N do
        loop (res) = for j < N do
            let partsum =
                let res = 0 in
                loop (res) = for k < N do
                    let res = res + a[i,k] * b[k,j]
                    in  res
                in res
            in let res[i,j] = partsum in res
        in res
    in res


fun [[int]] main([[int]] x, [[int]] y) =
  matmultImp(x, y)
