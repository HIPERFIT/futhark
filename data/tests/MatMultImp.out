Result of evaluation:  [  [ 19 , 22  ] ,  [ 43 , 50  ]  ]
